lib/improver/tests/nowcasting/pysteps_interfacing/test_PystepsImporter.py

```diff
@@ -0,0 +1,123 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for pysteps plotting interface"""
+
+import unittest
+import numpy as np
+
+from iris.tests import IrisTest
+
+from improver.nowcasting.pysteps_interfacing import PystepsImporter
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class Test__init__(IrisTest):
+    """Test the __init__ function"""
+
+    def test_basic(self):
+        """Test initialisation behaves as expected"""
+        plugin = PystepsImporter()
+        self.assertEqual(plugin.metadata['unit'], 'mm/h')
+        self.assertEqual(plugin.metadata['accutime'], 0)
+        self.assertIsNone(plugin.metadata['transform'])
+
+
+class Test_process_cube(IrisTest):
+    """Class to test the process_cube method"""
+
+    def setUp(self):
+        """Set up test cube to import"""
+        precip_data_ms = 1e-7*np.ones((5, 5), dtype=np.float32)
+        self.cube = set_up_variable_cube(
+            precip_data_ms, name='rainfall_rate', units='m s-1',
+            spatial_grid='equalarea', attributes={'institution': 'Met Office'})
+        self.precip_data_mmh = 0.36*np.ones((5, 5), dtype=np.float32)
+
+    def test_basic(self):
+        """Test outputs are of correct types"""
+        (data, metadata) = PystepsImporter().process_cube(self.cube)
+        self.assertIsInstance(data, np.ndarray)
+        self.assertIsInstance(metadata, dict)
+
+    def test_rainfall_values(self):
+        """Test array contains correct values in mm/h"""
+        (data, _) = PystepsImporter().process_cube(self.cube)
+        self.assertArrayAlmostEqual(data, self.precip_data_mmh)
+
+    def test_with_mask(self):
+        """Test plugin treats masked data correctly"""
+        mask = np.full((5, 5), False)
+        mask[:1, :] = True
+        mask[:, :1] = True
```

Comment:
Interesting syntax. Why not just ```mask[0, :]```?

Reply:
Blood caffeine level?  I guess I was thinking about where the data would be valid (ie cell 1) rather than where it wouldn't - this gave me a better picture of a grid with the edges cut off.  But I can change it if need be.